src/adapter/src/catalog.rs

```diff
@@ -6160,17 +6372,18 @@ impl Catalog {
         if new_present {
             // Group privileges by (grantee, grantor).
             let privilege_map: BTreeMap<_, Vec<_>> =
-                privileges
-                    .into_iter()
+                flat_privileges
+                    .drain(..)
```

Comment:
not positive on the types here, but on a `Vec` calling `.drain(..)` is the same as `.into_iter()`, and personally I'd prefer the later

Reply:
Updated